src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -1282,3 +1283,103 @@ def forward(
             hidden_states=outputs.hidden_states,
             attentions=outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+The Mistral Model transformer with a span classification head on top for extractive question-answering tasks like
+SQuAD (a linear layer on top of the hidden-states output to compute `span start logits` and `span end logits`).
+    """,
+    MISTRAL_START_DOCSTRING,
+)
+class MistralForQuestionAnswering(MistralPreTrainedModel):
+    # Copied from transformers.models.llama.modeling_llama.LlamaForQuestionAnswering.__init__ with Llama->Mistral,transformer->model
+    def __init__(self, config):
+        super().__init__(config)
+        self.model = MistralModel(config)
+        self.qa_outputs = nn.Linear(config.hidden_size, 2)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForQuestionAnswering.get_input_embeddings with transformer->model
+    def get_input_embeddings(self):
+        return self.model.embed_tokens
+
+    # Copied from transformers.models.llama.modeling_llama.LlamaForQuestionAnswering.set_input_embeddings with transformer->model
+    def set_input_embeddings(self, value):
+        self.model.embed_tokens = value
+
+    @add_start_docstrings_to_model_forward(MISTRAL_INPUTS_DOCSTRING)
+    # Copied from transformers.models.llama.modeling_llama.LlamaForQuestionAnswering.forward with Llama->Mistral, transformer->model
```

Comment:
Copying from each one individually due to llama having a wrong base model prefix which already led/leads to issues in the past: #30381. Currently, it makes copying force to include the base model prefix (ref. [here](https://github.com/huggingface/transformers/blob/a84c413773cdfdba58b35194c5ba51e2ccb2ca39/src/transformers/models/llama/modeling_llama.py#L1368)) if using top-level copied from.

Reply:
So it's more of a stylistic choice: individual copies vs. include (unnecessary) base model prefix.

Reply:
If #34061 gets merged, we can top-level copy from llama without any problems.

Reply:
You can also use `# Ignore copy` on the single place where the copy does not match!

Reply:
Ah ok, perfect I'll change it later and ping you when ready ;)